helix-core/src/main/java/org/apache/helix/task/AssignableInstanceManager.java

```diff
@@ -609,6 +611,19 @@ public void logQuotaProfileJSON(boolean onlyDisplayIfFull) {
     }
   }
 
+  /**
+   * For each JobType, record their total available threads across all instances
+   * @param clusterStatusMonitor
+   */
+  public void recordAvailableThreadsPerType(ClusterStatusMonitor clusterStatusMonitor) {
+    if (clusterStatusMonitor != null) {
```

Comment:
This kind of null check should ideally be done before `ClusterStatusMonitor` is passed in to `JobMonitor` to make the code more concise to read. See comment above :)